rust/src/combinators/composition/sequential/mod.rs

```diff
@@ -0,0 +1,240 @@
+                Ok(d_out.clone())
+            }
+        }),
+    ))
+}
+
+#[cfg(test)]
+mod test {
+
+    use std::any::Any;
+
+    use super::*;
+    use crate::{
+        domains::AllDomain, measurements::make_randomized_response_bool, measures::MaxDivergence,
+    };
+
+    #[test]
+    fn test_sequential_composition() -> Fallible<()> {
+        // construct sequential compositor IM
+        let root = make_sequential_composition::<_, Box<dyn Any>, _, _>(
+            AllDomain::new(),
+            MaxDivergence::default(),
+            1,
+            vec![0.1, 0.1, 0.3, 0.3, 0.5],
+        )?;
+
+        // pass dataset in and receive a queryable
+        let mut queryable = root.invoke(&true)?;
+
+        // construct the leaf-node queries:
+        let rr_poly_query = make_randomized_response_bool(0.5, false)?.into_poly();
+        let rr_query = make_randomized_response_bool(0.5, false)?;
+
+        // pass queries into the SC queryable
+        println!("the sequential compositor can be evaluated with rr poly queries, and the answer is downcast to bool");
+        let _answer1: bool = queryable.eval_poly(&rr_poly_query)?;
+        let _answer2: bool = queryable.eval_poly(&rr_poly_query)?;
+
+        println!("\nbuild a sequential composition IM and then convert to poly, so that it can be passed to the root queryable");
+        // pass a sequential composition compositor into the original SC compositor
+        // This compositor expects all outputs are concretely-typed (bool)
+        let sc_query_3 = make_sequential_composition::<_, bool, _, _>(
+            AllDomain::<bool>::new(),
+            MaxDivergence::default(),
+            1,
+            vec![0.1, 0.1],
+        )?
+        .into_poly();
+
+        // both approaches are valid
+        println!("\nAPPROACH A: submit poly queries to a poly queryable");
+        println!("\ncreate the sequential composition queryable as a child of the root queryable");
+        let mut answer3a = queryable.eval_poly::<Queryable<_, bool>>(&sc_query_3)?;
+
+        println!(
+            "\npass an RR query to the child sequential compositor queryable and get an answer"
+        );
+        let _answer3a_1: bool = answer3a.eval(&rr_query)?;
+
+        println!("\npass a second RR query to the child sequential compositor queryable");
+        let _answer3a_2: bool = answer3a.eval(&rr_query)?;
+
+        println!("\nAPPROACH B: downcast the poly queryable and then send normal queries");
+        println!("create the sequential composition queryable as a child of the root queryable, but downcast the queryable itself");
+        let mut answer3b = queryable.eval_poly::<Queryable<_, bool>>(&sc_query_3)?;
+
+        println!("\nsend a normal query without any dyn or poly type-erasure");
+        let _answer3b_1: bool = answer3b.eval(&rr_query)?;
+        println!("\nsend a second normal query without any dyn or poly type-erasure");
+        let _answer3b_2: bool = answer3b.eval(&rr_query)?;
```

Comment:
Both approaches seem the same. Are you missing the downcast of the queryable in Approach B?

Reply:
Ah, the approaches became the same after removing `into_any_queryable`.